### DISCUSSION TITLE
Machine Learning Milestone - Logistic Regression & SVM Completed

### DISCUSSION THREAD

**Discussion by saunakdas16** [saunakdas16/alzheimers-eeg-ml#2] [Announcements]
# 🚀 Machine Learning Stage Update — Logistic Regression & SVM Completed

A major milestone has been reached in my **EEG-Based Alzheimer's Disease Classification** project.

After completing the EEG preprocessing, artifact correction, ICA, spectral analysis, and subject-level feature extraction pipeline across the complete dataset of **88 participants**, I have now completed **two machine-learning classification experiments** for **Alzheimer's disease (AD) vs Cognitively Normal (CN)** participants.

## 🧮 Logistic Regression

The first classification model was **Logistic Regression**, used as a linear baseline for the binary AD vs CN classification task.

The model used five subject-level relative EEG spectral-power features:

- Delta
- Theta
- Alpha
- Beta
- Gamma

The dataset for the classification experiment contained:

- **36 AD participants**
- **29 CN participants**
- **65 participants total**

A stratified **80/20 train-test split** was used:

- **52 participants** for training
- **13 participants** for the held-out test set

The five EEG features were standardized using `StandardScaler` before fitting the model.

### Test-Set Results

**Test Accuracy:** 84.6%  
**ROC-AUC:** 0.976

The Logistic Regression analysis also included:

- Learned model coefficients
- Prediction-confidence visualization
- Feature-contribution visualization

The coefficient analysis provides an interpretable view of the direction and relative magnitude of the standardized EEG features within the linear model.

## ⚙️ Support Vector Machine

The second classification experiment used a **Support Vector Machine (SVM)** with an **RBF kernel** to model nonlinear decision boundaries.

The same **65-participant AD vs CN dataset** and the same **52/13 train-test split** were used so that the held-out test set remained consistent between the two experiments.

### Initial SVM

The initial model used the default `SVC()` configuration:

```text
Kernel = RBF
C = 1.0
Gamma = scale
```

Initial test-set performance:

**Test Accuracy:** 76.9%

**ROC-AUC:** 0.952

### Cross-Validation & Hyperparameter Tuning

For the SVM, **5-fold stratified cross-validation** was performed only within the **52 training participants**.

The initial cross-validation accuracy was:

**Mean Accuracy:** 77.3%

**Standard Deviation:** 8.8%

`GridSearchCV` was then used to evaluate:

```text
C = 0.1, 1, 10, 100
Gamma = scale, 0.01, 0.1, 1
```

This resulted in **16 parameter combinations**, each evaluated using 5-fold stratified cross-validation within the training set.

### Best SVM Parameters

The selected parameters were:

```text
C = 10
Gamma = 0.01
```

**Best Mean Cross-Validation Accuracy:** 81.1%

**Cross-Validation Standard Deviation:** 7.7%

### Tuned SVM Test Results

The tuned SVM was then evaluated on the same untouched **13-participant test set**.

**Test Accuracy:** 76.9%

**ROC-AUC:** 1.000

The tuned SVM produced the same class predictions as the initial SVM on the held-out test set, resulting in:

**10 of 13 test participants correctly classified.**

The ROC-AUC of 1.000 reflects the ranking of decision scores on this particular small test set and should not be interpreted as evidence of perfect generalization.

## 📊 Model Visualizations

Both Logistic Regression and SVM now have corresponding visualizations covering:

**Prediction Confidence → Feature Interpretation → Feature Contributions**

These visualizations provide different perspectives on how the two models use the EEG spectral features.

A dedicated **Logistic Regression vs SVM** comparison visualization was also created using the recorded test-set results from both models.

The comparison includes:

- Test Accuracy
- ROC-AUC

Both models were evaluated on the same **13-participant held-out test set**.

## 🔒 Methodological & Data Leakage Considerations

An important methodological refinement during this stage was identifying the difference between the initial Logistic Regression cross-validation procedure and the corrected SVM workflow.

The initial Logistic Regression cross-validation was performed on all **65 AD/CN participants**, which included the subjects from the held-out test set. Therefore, those cross-validation results are treated as **exploratory** rather than as an independent validation estimate.

For the SVM, cross-validation and hyperparameter tuning were performed only within the **52 training participants**.

Feature scaling was also included inside the SVM pipeline so that scaling parameters were learned independently within each training fold.

The final **13-participant test set remained untouched during SVM model selection and hyperparameter tuning**.

Future machine-learning experiments will follow this corrected subject-level evaluation approach.

## 🧠 Learnings From This Stage

This stage was not only about training models, but also about understanding the importance of a proper machine-learning workflow for EEG data.

Some of the key concepts explored during this stage include:

- Binary classification
- Train-test splitting
- Feature standardization
- Logistic Regression
- Support Vector Machines
- RBF kernels
- Cross-validation
- Hyperparameter tuning
- `GridSearchCV`
- Confusion matrices
- Classification metrics
- ROC curves and ROC-AUC
- Prediction confidence
- Feature importance
- Feature contributions
- Data leakage and subject-level separation

## 🔜 What's Next?

The next classification experiment will focus on a **Random Forest** model using the same subject-level EEG feature dataset.

The goal is to continue building and comparing different machine-learning approaches while maintaining a consistent subject-level evaluation framework.

This project is being developed step by step as both a **research-oriented EEG analysis project and a learning journey in computational neuroscience, signal processing, and machine learning**.

🔗 **Repository:**

https://github.com/saunakdas16/alzheimers-eeg-ml
